migration/sql-files/006-work-item-links.sql

```diff
@@ -0,0 +1,89 @@
+-- Here's the layout I'm trying to create:
+-- (NOTE: work_items and work_item_types tables already exist)
+--        
+--           .----------------.
+--           | work_items     |        .-----------------.
+--           | ----------     |        | work_item_types |
+--     .------>id bigserial   |        | --------------- |
+--     |     | type text ------------>>| name text       |
+--     |     | [other fields] |    |   | [other fields]  |
+--     |     '----------------'    |   '-----------------'
+--     |                           |
+--     |   .------------------.    |   .-----------------------.
+--     |   | work_item_links  |    |   | work_item_link_types  |
+--     |   | ---------------  |    |   | ---------             |
+--     |   | id uuid          | .------> id uuid               |
+--     .-----source_id bigint | |  |   | name text             |
+--      '----target_id bigint |/   |   | description text      |
+--         | link_type_id uuid|    '-----source_type_name text |
+--         '------------------'     '----target_type_name text |
+--                                     | forward_name text     |
+--                                     | reverse_name text     |
+--    .--------------------------.    .- link_category_id uuid |
+--    |work_item_link_categories |   / '-----------------------'
+--    |------------------------- |  /
+--    | id uuid                 <---
+--    | name text                |
+--    | description text         |
+--    '--------------------------'
+
+CREATE EXTENSION IF NOT EXISTS "uuid-ossp";
+
+-- work item link categories
+
+CREATE TABLE work_item_link_categories (
+    created_at  timestamp with time zone,
+    updated_at  timestamp with time zone,
+    deleted_at  timestamp with time zone DEFAULT NULL,
+
+    id          uuid primary key DEFAULT uuid_generate_v4() NOT NULL,
+    version     integer DEFAULT 0 NOT NULL,
+
+    name        text NOT NULL,
+    description text
+);
+
+-- Ensure we only have one link category with the same name in existence.
+-- If a category has been deleted (deleted_at != NULL) then we can recreate the category with the same name again.
+CREATE UNIQUE INDEX work_item_link_categories_name_idx ON work_item_link_categories (name) WHERE deleted_at IS NULL;
+
+-- work item link types
+
+CREATE TYPE work_item_link_topology AS ENUM ('network', 'directed_network', 'dependency', 'tree');
```

Comment:
Am I correct in assuming that these names correspond to "unconstraind graph", "directed graph", "directed acyclic graph" and "tree"?
I'm not sure what the most clear and widely understood terminology here would be, but I think it is worth naming these constants well. People will infer things from these names. 
https://en.wikipedia.org/wiki/Graph_(discrete_mathematics)#Types_of_graphs might be a start.